### PR TITLE
Add permissions selection section to the create role interface

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose
Added permissions selection section to the create role interface

### Screenshot
Screenshot 2023-10-22 at 15 11 32

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/roles/components/edit-role/edit-role-common/render-chip.tsx [hunk 1]
```diff
@@ -32,23 +30,27 @@ interface RenderChipInterface extends IdentifiableComponentInterface, ChipProps
     /**
      * Callback to set the active option.
      */
-    setActiveOption: (option: GroupsInterface|UserBasicInterface) => void;
+    setActiveOption: (option: any) => void;
```
Comment (DonOmalVindula) [inline apps/console/src/features/roles/components/edit-role/edit-role-common/render-chip.tsx:33]: Can we avoid using `any` here if possible?

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/components/permissions-list.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com).
```
Comment (DonOmalVindula) [inline apps/console/src/features/roles/components/wizard-updated/role-permissions/components/permissions-list.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/components/role-api-resources-list-item.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com).
```
Comment (DonOmalVindula) [inline apps/console/src/features/roles/components/wizard-updated/role-permissions/components/role-api-resources-list-item.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/components/role-api-resources-list-item.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    selectedPermissions: ScopeInterface[];
+    /**
+     * Callback to handle API resource removal.
+     */
+    onChangeScopes: (apiResource: APIResourceInterface, scopes: ScopeInterface[]) => void;
+    /**
+     * Callback to handle API resource removal.
+     */
+    onRemoveAPIResource: (apiResourceId: string) => void;
+
```
Comment (DonOmalVindula) [inline apps/console/src/features/roles/components/wizard-updated/role-permissions/components/role-api-resources-list-item.tsx:50]:
```suggestion
```

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx [hunk 5]
```diff
@@ -0,0 +1,278 @@
+/**
+ * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com).
```
Comment (DonOmalVindula) [inline apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+                if (!selectedAPIResources.find(
+                    (apiResource: APIResourceInterface) => selectedAPIResource?.id === apiResource?.id)) {
+                    setSelectedAPIResources([ selectedAPIResource, ...selectedAPIResources ]);
+                }
+
+                selectedAPIResourceId && setSelectedAPIResourceId(undefined);
+            }
+        }, [ selectedAPIResource ]);
+
+        // TODO: Need to handle the error state.
```
Comment (DonOmalVindula) [inline apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx:141]: Is this intentional? is there a blocker to handle the error state?

File: apps/console/src/features/roles/models/apiResources.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2020-2023, WSO2 LLC. (https://www.wso2.com).
```
Comment (DonOmalVindula) [inline apps/console/src/features/roles/models/apiResources.ts:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```

**Review by DonOmalVindula — DISMISSED** [review]

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx (hunk 6 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx:141]: Error state is handled, will remove the comment

File: apps/console/src/features/roles/components/edit-role/edit-role-common/render-chip.tsx (hunk 1 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-common/render-chip.tsx:33]: Here we are passing three different Interfaces `GroupsInterface`, `UserBasicInterface` and `ScopeInterface` hence will require to use `any`.

File: apps/console/src/features/roles/models/apiResources.ts (hunk 7 above)
Comment (Achintha444) [reply apps/console/src/features/roles/models/apiResources.ts]: Fixed with https://github.com/wso2/identity-apps/pull/4278/commits/dc1ed8886e3e85f06247f47bbcf6eae95045a9bd

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx (hunk 6 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4278/commits/dc1ed8886e3e85f06247f47bbcf6eae95045a9bd

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx (hunk 5 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4278/commits/dc1ed8886e3e85f06247f47bbcf6eae95045a9bd

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/components/role-api-resources-list-item.tsx (hunk 4 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/wizard-updated/role-permissions/components/role-api-resources-list-item.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4278/commits/dc1ed8886e3e85f06247f47bbcf6eae95045a9bd

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/components/role-api-resources-list-item.tsx (hunk 3 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/wizard-updated/role-permissions/components/role-api-resources-list-item.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4278/commits/dc1ed8886e3e85f06247f47bbcf6eae95045a9bd

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/components/permissions-list.tsx (hunk 2 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/wizard-updated/role-permissions/components/permissions-list.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4278/commits/dc1ed8886e3e85f06247f47bbcf6eae95045a9bd

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+            isLoading: isAPIResourcesListFetchRequestLoading,
+            error: apiResourcsListFetchRequestError,
+            mutate: mutateAPIResourcesListFetchRequest
+        } = useAPIResourcesList(apiResourceSearchQuery);
+
+        const {
+            data: selectedAPIResource,
+            isLoading: isSelectedAPIResourceFetchRequestLoading,
+            isValidating: isSelectedAPIResourceFetchRequestValidating,
+            error: selectedAPIResourceFetchRequestError
```
Comment (JayaShakthi97) [inline apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx:84]:
```suggestion
            error: selectedAPIResourceFetchRequestError
```

**Review by DonOmalVindula — DISMISSED** [review]

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+                    setAPIResourceSearchQuery(undefined);
+                });
+            }, RoleConstants.DEBOUNCE_TIMEOUT), []);
+
+        /**
+         * Handles the change of the search query of application list.
+         */
+        const onSearchChangeAPIResourcs = (event: SyntheticEvent<HTMLElement>, data: DropdownProps): void => {
+            setAPIResourcesSearching(true);
+            searchAPIResources(data.searchQuery.toString().trim());
```
Comment (JayaShakthi97) [inline apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx:158]: Any chance of `searchQuery` being null or undefined?

**Review by JayaShakthi97 — APPROVED** [review]

File: apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx (hunk 9 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4278/commits/445e6e01208a111c7632813406802eb382873a7f

**Review by JayaShakthi97 — APPROVED** [review]

File: apps/console/src/features/roles/components/edit-role/edit-role-common/render-chip.tsx (hunk 1 above)
Comment (dasuni-30) [reply apps/console/src/features/roles/components/edit-role/edit-role-common/render-chip.tsx:33]: Can we use | and add all the interfaces?

**Review by dasuni-30 — APPROVED** [review]

**Pull request merged**